jax_galsim/interpolant.py

```diff
@@ -0,0 +1,1679 @@
+            _C = tuple(_C)
+            self._K = _K
+            self._C = _C
+        else:
+            self._K = _K
+            self._C = _C
+
+        self._K_arr = jnp.array(self._K, dtype=float)
+        self._C_arr = jnp.array(self._C, dtype=float)
+
+        if _du is None:
+            _du = (
+                self._gsparams.table_spacing
+                * jnp.power(self._gsparams.kvalue_accuracy / 200.0, 0.25)
+                / self._n
+            ).item()
+            self._du = _du
+        else:
+            self._du = _du
+
+        if _umax is None:
+            self._umax = _find_umax_lanczos(
+                self._du,
+                self._n,
+                self._conserve_dc,
+                self._C,
+                self._gsparams.kvalue_accuracy,
+            ).item()
+        else:
+            self._umax = _umax
+
+    def tree_flatten(self):
+        """This function flattens the Interpolant into a list of children
+        nodes that will be traced by JAX and auxiliary static data."""
+        # Define the children nodes of the PyTree that need tracing
+        children = tuple()
+        # Define auxiliary static data that doesn’t need to be traced
+        aux_data = {
+            "gsparams": self._gsparams,
+            "n": self._n,
+            "conserve_dc": self._conserve_dc,
+        }
+        if hasattr(self, "_du"):
+            aux_data["_du"] = self._du
+        if hasattr(self, "_umax"):
+            aux_data["_umax"] = self._umax
+        if hasattr(self, "_K"):
+            aux_data["_K"] = self._K
+            aux_data["_C"] = self._C
+        return (children, aux_data)
+
+    @classmethod
+    def tree_unflatten(cls, aux_data, children):
+        """Recreates an instance of the class from flattened representation"""
+        n = aux_data.pop("n")
+        return cls(n, **aux_data)
+
+    def __repr__(self):
+        return "galsim.Lanczos(%r, %r, gsparams=%r)" % (
+            self._n,
+            self._conserve_dc,
+            self._gsparams,
+        )
+
+    def __str__(self):
+        return "galsim.Lanczos(%s)" % (self._n)
+
+    # this is a pure function and we apply JIT ahead of time since this
+    # one is pretty slow
+    @jax.jit
+    def _xval(x, n, conserve_dc, _K):
+        x = jnp.abs(x)
+
+        def _low(x, n):
+            # from galsim
+            # // res = n/(pi x)^2 * sin(pi x) * sin(pi x / n)
+            # //     ~= (1 - 1/6 pix^2) * (1 - 1/6 pix^2 / n^2)
+            # //     = 1 - 1/6 pix^2 ( 1 + 1/n^2 )
+            # // For x < 1.e-4, the errors in this approximation are less than 1.e-16.
+            pix = jnp.pi * x
+            temp = (1.0 / 6.0) * pix * pix
+            return 1.0 - temp * (1.0 + 1.0 / (n * n))
+
+        def _high(x, n):
+            pix = jnp.pi * x
+            s = jnp.sin(pix)
+            sn = jnp.sin(pix / n)
+            return n * s * sn / (pix * pix)
+
+        msk_s = x <= 1e-4
+        msk_l = x <= n
+        val = jnp.piecewise(
```

Comment:
I see you didn't specialize the `n` cases like in [Galsim](https://github.com/GalSim-developers/GalSim/blob/a1f0a11a24690a279abd7027b116ec3e4986643d/src/Interpolant.cpp#L579), do we loss significant speed this way?

Reply:
I didn't test that and plan to defer that work to future PRs. The goal here is to get something working.